src/vstart.sh

```diff
@@ -1358,6 +1366,12 @@ osd pool create ec erasure ec-profile
 EOF
 fi
 
+# Avoid issues with max number of PGs per OSD when creating RGW buckets.
+# See: https://github.com/ceph/ceph/pull/33691#pullrequestreview-367974456
+if [ $GANESHA_DAEMON_NUM -gt 0 ] || [ "$CEPH_NUM_RGW" -gt 0 ]; then
+    ceph_adm config set global mon_max_pg_per_osd ${MON_MAX_PG_PER_OSD:-1000}
+fi
```

Comment:
i'd suggest move this hunk into `do_rgw_conf()` like
```diff
diff --git a/src/vstart.sh b/src/vstart.sh
index 926b8a1622..3202faa65a 100755
--- a/src/vstart.sh
+++ b/src/vstart.sh
@@ -535,7 +535,18 @@ do_rgw_conf() {
         admin socket = ${CEPH_OUT_DIR}/radosgw.${current_port}.asok
 EOF
         current_port=$((current_port + 1))
-done
+    done
+    # Avoid issues with max number of PGs per OSD when creating RGW buckets.
+    # See: https://github.com/ceph/ceph/pull/33691#pullrequestreview-367974456
+    local mon_max_pg_per_osd=${MON_MAX_PG_PER_OSD:-1000}
+    if [ $GANESHA_DAEMON_NUM -gt 0 ]; then
+        wconf <<EOF
+[mon]
+mon_max_pg_per_osd ${mon_max_pg_per_osd}
+[osd]
+mon_max_pg_per_osd ${mon_max_pg_per_osd}
+EOF
+    fi

 }
```

Reply:
@tchaikov  I tried your suggestion but it didn't work as I faced this error:
```
rgw_init_ioctx ERROR: librados::Rados::pool_create returned (34) Numerical result out of range (this can be due to a pool or placement group misconfiguration, e.g. pg_num < pgp_num or mon_max_pg_per_osd exceeded)
```
So I prefer to keep it as it is.

Reply:
@alfonsomthd i think i forgot to add "=" in between `mon_max_pg_per_osd` and `${mon_max_pg_per_osd}`.

did you add the "=" when testing?

Reply:
@tchaikov I tested again (ensuring "=" was there) but getting the same error when multiple pools created: i.e. enabling ganesha + multisite (2 realms, 2 zonegroups per realm, 2 zones per zonegroup).

Reply:
@tchaikov I found that it was a misconfiguration from my side.
But after our IRC/BJ conversation, I applied the setting globally (see updated PR description)
for you to review & test.

Reply:
thanks ! i also checked the tests for this option, they are 

- `qa/suites/rados/singleton/all/max-pg-per-osd.from-primary.yaml`
- `qa/suites/rados/singleton/all/max-pg-per-osd.from-mon.yaml`
- `qa/suites/rados/singleton/all/max-pg-per-osd.from-replica.yaml`

and what we are using when performing tests is in `qa/tasks/ceph.conf.template`.

like

```
$ git grep 'mon max pg per osd'
doc/rados/configuration/pool-pg-config-ref.rst:              ``osd max pg per osd hard ratio`` \* ``mon max pg per osd``.
qa/suites/rados/singleton/all/max-pg-per-osd.from-mon.yaml:        mon max pg per osd : 2
qa/suites/rados/singleton/all/max-pg-per-osd.from-primary.yaml:        mon max pg per osd : 1
qa/suites/rados/singleton/all/max-pg-per-osd.from-replica.yaml:        mon max pg per osd : 1
qa/tasks/ceph.conf.template:    mon max pg per osd = 10000        # >= luminous
qa/tasks/cephadm.conf:mon max pg per osd = 10000        # >= luminous
```